local a = hotki.actions
local relay = a.relay_with("ctrl+shift+cmd")

return function(menu: MenuBuilder, _ctx: ModeContext)
    menu:submenu("shift+cmd+j", "obsidian", function(app_menu)
        app_menu:submenu("c", "content", function(content)
            content:submenu("i", "insert", function(insert)
                insert:bind("a", "attachment", relay("6"))
                insert:bind("c", "code", relay("7"))
                insert:bind("f", "footnote", relay("8"))
                insert:bind("l", "link", relay("9"))
            end)
            content:submenu("t", "table", function(tbl)
                local stay = tbl:with({ stay = true })
                stay:bind("n", "new", relay("y"))
                tbl:submenu("i", "insert", function(insert)
                    local stay = insert:with({ stay = true })
                    stay:bind("h", "column left", relay("u"))
                    stay:bind("l", "column right", relay("i"))
                    stay:bind("j", "column above", relay("o"))
                    stay:bind("k", "column below", relay("p"))
                end)
                tbl:submenu("d", "delete", function(delete_menu)
                    local stay = delete_menu:with({ stay = true })
                    stay:bind("r", "row", relay("m"))
                    stay:bind("c", "column", relay("n"))
                end)
            end)
        end)
        app_menu:submenu("f", "file", function(file)
            file:bind("f", "show in finder", relay("4"))
            file:bind("n", "rename", relay("5"))
        end)
        app_menu:submenu("s", "split", function(split)
            split:bind("l", "right", relay("2"))
            split:bind("j", "down", relay("1"))
        end)
        app_menu:submenu("u", "utils", function(utils)
            utils:bind("s", "shell here", relay("s"))
        end)
        app_menu:submenu("w", "window", function(window)
            window:bind("h", "sidebar left", relay("h"))
            window:bind("l", "sidebar right", relay("l"))
            window:bind("s", "settings", relay("3"))
        end)
    end)
end
